crates/bevy_ecs/src/archetype.rs

```diff
@@ -317,8 +317,8 @@ pub struct ArchetypeGeneration(usize);
 
 impl ArchetypeGeneration {
     #[inline]
-    pub fn new(generation: usize) -> Self {
-        ArchetypeGeneration(generation)
+    pub fn initial() -> Self {
+        ArchetypeGeneration(0)
```

Comment:
Is there a reason we're using 0 instead of `usize::MAX` like the previous implementation did?

Also the function could be marked `const`

Reply:
This is pretty much the point of this PR - simplify the initialization logic by not making MAX a special-case initial value, but use 0 value directly instead.